### PR TITLE
salt: store CA minion in `BootstrapConfiguration`

### PR DESCRIPTION
**Pull request by NicolasT**
Add a new field in the `BootstrapConfiguration` to specify the Salt
minion to use as a CA server, instead of mine-based guesswork.

Also remove the checks for 'existing CAs', which don't really make
sense: in a way, multiple servers may be in a CA 'state', as long as
only one of them (in our case, the one from the configuration) is being
used.

Over time, we could implement an orchestration playbook to 'migrate' a
CA from one node to another.

Fixes: #632
See: https://github.com/scality/metalk8s/issues/632

### CODE REVIEW

**Comment by NicolasT** [comment]
@bert-e reset

File: Vagrantfile [hunk 1]
```diff
@@ -119,11 +119,13 @@ fi
 
 echo "Creating bootstrap configuration"
 cat > /etc/metalk8s/bootstrap.yaml << EOF
-apiVersion: metalk8s.scality.com/v1alpha1
+apiVersion: metalk8s.scality.com/v1alpha2
```
Comment (slaperche-scality) [inline Vagrantfile:122]: All those `s/v1alpha1/v1alpha2/` seems unrelated to the CA business, so they belong to a dedicated commit or PR.

**Review by slaperche-scality — CHANGES_REQUESTED** [review]
Ho, lot of cruft removed in this one \o/

File: Vagrantfile (hunk 1 above)
Comment (NicolasT) [reply Vagrantfile:122]: Huh? The CA is now in the configuration file, hence bumping the version.

Comment (slaperche-scality) [reply Vagrantfile:122]: Ho, OK.

I thought it was related to the release of the alpha2.

**Review by slaperche-scality — DISMISSED** [review]

**Comment by NicolasT** [comment]
@bert-e reset

**Review by slaperche-scality — DISMISSED** [review]

File: Vagrantfile (hunk 1 above)
Comment (NicolasT) [reply Vagrantfile:122]: No no :) It's purely about the 'version' of this configuration file. In a sense, could've added the new field without bumping the version, we're nowhere near something 'frozen' anyway, but I think it made sense to set an example.

Either way, definitely not related to the MetalK8s version.

**Comment by NicolasT** [comment]
@bert-e reset

**Review by sayf-eddine-scality — DISMISSED** [review]
still used in
`metalk8s/kubeadm/init/kubeconfig/lib.sls`
`metalk8s.kubeadm.init.etcd.local`

**Comment by NicolasT** [comment]
@bert-e reset

**Comment by NicolasT** [comment]
@bert-e reset

**Comment by NicolasT** [comment]
@bert-e reset

**Comment by NicolasT** [comment]
@bert-e reset

File: salt/metalk8s/kubeadm/init/certs/ca.sls [hunk 2]
```diff
@@ -38,8 +33,7 @@ Generate CA certificate:
     - require:
       - x509: Create CA private key
 
-# TODO: Find a better way to advetise CA server
-Advertise CA in the mine:
+Advertise CA certificate in the mine:
```
Comment (TeddyAndrieux) [inline salt/metalk8s/kubeadm/init/certs/ca.sls:36]: This one (and the other CA) should be a `mine_function` in the pillar automaticaly updated with `mine.update` because if we flush the mine we will need to advertise again to the mine manually.

But could be done in another PR/ticket btw

File: salt/metalk8s/kubeadm/init/certs/etcd-ca.sls [hunk 3]
```diff
@@ -38,7 +33,6 @@ Generate etcd CA certificate:
     - require:
       - x509: Create etcd CA private key
 
-# TODO: Find a better way to advertise CA server
 Advertise etcd CA in the mine:
```
Comment (TeddyAndrieux) [inline salt/metalk8s/kubeadm/init/certs/etcd-ca.sls]: To be consistent with the previous change for basic CA
```suggestion
Advertise etcd CA certificate in the mine:
```

File: salt/metalk8s/kubeadm/init/certs/front-proxy-ca.sls [hunk 4]
```diff
@@ -38,7 +33,6 @@ Generate front proxy CA certificate:
     - require:
       - x509: Create front proxy CA private key
 
-# TODO: Find a better way to advertise CA server
 Advertise front proxy CA in the mine:
```
Comment (TeddyAndrieux) [inline salt/metalk8s/kubeadm/init/certs/front-proxy-ca.sls]:
```suggestion
Advertise front proxy CA certificate in the mine:
```

File: salt/metalk8s/kubeadm/init/certs/etcd-ca.sls (hunk 3 above)
Comment (TeddyAndrieux) [reply salt/metalk8s/kubeadm/init/certs/etcd-ca.sls]: (strange that github suggestion remove a space not existing at the beginning of the line)

File: eve/main.yml [hunk 5]
```diff
@@ -246,6 +248,8 @@ stages:
             ssh -F bootstrap_ssh_config bootstrap
             sudo mkdir -p /etc/metalk8s &&
             scp -F bootstrap_ssh_config bootstrap.yaml bootstrap: &&
+            ssh -F bootstrap_ssh_config
```
Comment (sayf-eddine-scality) [inline eve/main.yml]: hmm I guess eve doesnt like this line break
```
usage: ssh [-1246AaCfGgKkMNnqsTtVvXxYy] [-b bind_address] [-c cipher_spec]
           [-D [bind_address:]port] [-E log_file] [-e escape_char]
           [-F configfile] [-I pkcs11] [-i identity_file]
           [-J [user@]host[:port]] [-L address] [-l login_name] [-m mac_spec]
           [-O ctl_cmd] [-o option] [-p port] [-Q query_option] [-R address]
           [-S ctl_path] [-W host:port] [-w local_tun[:remote_tun]]
           [user@]hostname [command]
/bin/sh: line 1: sh -c "sed -i s/@HOSTNAME@/$(hostname)/ boostrap.yaml": No such file or directory
```

File: eve/main.yml [hunk 6]
```diff
@@ -246,6 +248,8 @@ stages:
...
+                'sh -c "sed -i s/@HOSTNAME@/$(hostname)/ boostrap.yaml"' &&
```
Comment (gdemonet) [inline eve/main.yml]: The error probably comes from the typo:

```suggestion
                'sh -c "sed -i s/@HOSTNAME@/$(hostname)/ bootstrap.yaml"' &&
```

File: salt/metalk8s/kubeadm/init/certs/ca.sls (hunk 2 above)
Comment (NicolasT) [reply salt/metalk8s/kubeadm/init/certs/ca.sls:36]: That should be another ticket indeed.

**Comment by NicolasT** [comment]
@bert-e reset

File: salt/metalk8s/kubeadm/init/certs/ca.sls (hunk 2 above)
Comment (TeddyAndrieux) [reply salt/metalk8s/kubeadm/init/certs/ca.sls:36]: Created #960

**Review by TeddyAndrieux — DISMISSED** [review]
Code LGTM

**Comment by NicolasT** [comment]
@bert-e reset

**Comment by NicolasT** [comment]
@bert-e reset

**Comment by NicolasT** [comment]
@bert-e reset

**Comment by NicolasT** [comment]
@bert-e reset

**Comment by NicolasT** [comment]
@bert-e reset

**Comment by NicolasT** [comment]
@bert-e reset

**Comment by NicolasT** [comment]
@bert-e approve

**Review by TeddyAndrieux — APPROVED** [review]
